client/settings/payment-settings/__tests__/general-settings-section.test.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import GeneralSettingsSection from '../general-settings-section';
+
+describe( 'GeneralSettingsSection', () => {
+	it( 'should enable stripe when stripe checkbox is clicked', () => {
+		render( <GeneralSettingsSection /> );
+
+		expect( screen.getByLabelText( 'Enable Stripe' ) ).not.toBeChecked();
```

Comment:
With `styled`, I tried doing the following but test would fail:
```
const enableStripe = screen.getByLabelText( 'Enable Stripe' );
const enableTestMode = screen.getByLabelText( 'Enable test mode' );

userEvent.click( enableStripe );

expect( enableStripe ).not.toBeChecked();
expect( enableTestMode ).not.toBeChecked();
```

I ended up needing to query the screen again to on the checkbox status, otherwise it would reference the old checkbox status. Seems like it could be rendering related.

Reply:
This is expected I think. The constant `enableStripe` holds the initial state of the element. Any change in the dom or rendering does not change/reassign the value in `enableStripe`.  

The way you did it seems to be the correct approach. :+1: